src/components/side_panel/chart/chart_with_axis/design_panel.ts

```diff
@@ -65,7 +76,8 @@ export class ChartWithAxisDesignPanel extends Component<Props, SpreadsheetChildE
     if (!runtime || !("chartJsConfig" in runtime)) {
       return [];
     }
-    return runtime.chartJsConfig.data.datasets.map((d) => d.label);
+    const datasets = runtime.chartJsConfig.data.datasets;
+    return this.props.definition.dataSets.map((d, i) => datasets[i].label);
```

Comment:
This seems very error prone, that will break if we ever add trendlines between datasets :x